plugins/techdocs-react/src/hooks.ts

```diff
@@ -82,6 +82,8 @@ export const useShadowRootSelection = (waitMillis: number = 0) => {
     [shadowRoot, setSelection, waitMillis],
   );
 
+  useEffect(() => handleSelectionChange.cancel, [handleSelectionChange.cancel]);
```

Comment:
It's not calling cancel now. And it does this stuff every single time that cancel is changed, so it'll immediately cancel always, unconditionally? What's this thing meaning to achieve really 🤔

Reply:
So what I really want is for the cancel function to be called when the component unmounts. It was previously setting the state of an unmounted component.

Since it is returned by the effect setup function,`cancel` will be used as the [cleanup function](https://react.dev/reference/react/useEffect#parameters), and while it is indeed called on every render (which is not necessarily what I want), it is called when the component unmounts. Since the site typically only renders once, and canceling the debounce isn't that problematic in this case, I thought it was acceptable. Aand I don't really know any other way to achieve this.

If there is a better way I'm of course happy to do that!

Reply:
Oh, right, I kinda should have gotten that! But it also highlights that the code might be "too clever" and may mislead the reader. This makes it more explicit:

```ts
useEffect(() => {
  // cleanup on unmount
  return () => handleSelectionChange.cancel();
}, [handleSelectionChange.cancel]);
```

Reply:
Yeah, fair point. Updated!